File: client/java/api/src/com/mxb360/eregec/Eregec.java
package com.mxb360.eregec;

import com.mxb360.http.HttpClient;

public class Eregec {
    private HttpClient loginHttpClient = null;
    private String errorMessage = "";

    private String host = null;
    private LoginJsonResult loginJsonResult;

    public Eregec(String host) {
        this.host = host;

        loginHttpClient = new HttpClient(host + "/eregec/api/login");
    }

    public boolean login(String userName, String password) {
        loginHttpClient.clearParam();
        loginHttpClient.addParam("name", userName);
        loginHttpClient.addParam("password", password);
        String res = loginHttpClient.post();
        if (res.equals(""))
            errorMessage = loginHttpClient.getErrorMessage();

        loginJsonResult = new LoginJsonResult(res, errorMessage);
        errorMessage = loginJsonResult.isOk() ? "no error" : loginJsonResult.getErrorMessage();
        return loginJsonResult.isOk();
    }

    public boolean isLogin() {
        return loginJsonResult != null && loginJsonResult.isOk();
    }

    public LoginJsonResult getLoginJsonResult() {
        return loginJsonResult;
    }

    public String getUserId() {
        return isLogin() ? loginJsonResult.getUserId() : "";
    }

    public String getErrorMessage() {
        return errorMessage;
    }
}

File: client/python/eregec.py
class UserInformation:
    def __init__(self):
        pass

class PlatformInformation:
    def __init__(self):
        pass

class Eregec:
    def __init__(self, host):
        self.host = "http://" + host
        self.errorMessage = ""

    def login(self, userName, password):
        return False

    def logout(self):
        pass

    def getUserId(self):
        return None

    def getErrorMessage(self):
        return self.errorMessage

    def getIntegerPlatformData(self, name):
        return None

    def getFloatPlatformData(self, name):
        return None

    def getStringPlatformData(self, name):
        return None

    def getPlatformInformation(self): 
        return None

    def getUserInformation(self):
        return None

    def isLogin(self):
        return False

    def sendCommand(self, command):
        return False

    def downloadPlatformData(self):
        return False
